### PR TITLE
dwarf-fortress-packages.dwarf-fortress_0_47_02: init

### PR DESCRIPTION
**Pull request by shazow**
###### Motivation for this change

Bugfix release for Dwarf Fortress came out. This PR updates the themes.json and games.json (and the update script that updates games.json).

Unfortunately there's no corresponding unfuck release yet, so it's still using the 0.47.01 version of unfuck for 0.47.02 which seems to work fine (as it's a minor bugfix release).

Currently playing this version with:

```
nix-env -f . -iE 'p: (p {}).dwarf-fortress-packages.dwarf-fortress_0_47_02.override { enableTruetype = true; theme = "phoebus"; }'
```

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by shazow** [comment]
Hm, I might have missed something with unfuck, but not quite sure what.

**Comment by shazow** [comment]
Ah looks like unfuck breaks on trying to install the older `dwarf-fortress_0_47_01`. Hm, my nix-fu is not quite good enough to figure out what's going on, pointers appreciated.

File: pkgs/games/dwarf-fortress/update.sh [hunk 1]
```diff
@@ -8,8 +8,8 @@ systems='linux linux32 osx osx32
 
 if [ $# -eq 0 ]; then
     versions="$(curl http://www.bay12games.com/dwarves/ \
-		   | grep 'DOWNLOAD DWARF FORTRESS' \
-		   | sed 's/.*DOWNLOAD DWARF FORTRESS \([0-9.]*\) .*/\1/')"
+		   | grep 'DWARF FORTRESS CLASSIC ' \
+		   | sed 's/.*DWARF FORTRESS CLASSIC \([0-9.]*\) .*/\1/')"
 else
```
Comment (jonringer) [inline pkgs/games/dwarf-fortress/update.sh]: I just entered the version explicitly last time, thanks!

**Review by jonringer — APPROVED** [review]
LGTM

**Comment by jonringer** [comment]
unfortunately, this is does fail without a compatible version of unfuck:
```
while evaluating the attribute 'dfInit' of the derivation 'dwarf-fortress-0.47.01' at /home/jon/.cache/nixpkgs-review/pr-79630/nixpkgs/pkgs/games/dwarf-fortress/wrapper/default.nix:84:3:
while evaluating the attribute 'env' of the derivation 'dwarf-fortress-init' at /home/jon/.cache/nixpkgs-review/pr-79630/nixpkgs/pkgs/games/dwarf-fortress/wrapper/default.nix:87:5:
while evaluating the attribute 'passAsFile' of the derivation 'dwarf-fortress-env-0.47.01' at /home/jon/.cache/nixpkgs-review/pr-79630/nixpkgs/pkgs/build-support/trivial-builders.nix:7:14:
while evaluating the attribute 'buildCommand' of the derivation 'fixup' at /home/jon/.cache/nixpkgs-review/pr-79630/nixpkgs/pkgs/build-support/trivial-builders.nix:7:14:
while evaluating the attribute 'installPhase' of the derivation 'dwarf-fortress-0.47.01' at /home/jon/.cache/nixpkgs-review/pr-79630/nixpkgs/pkgs/games/dwarf-fortress/game.nix:46:3:
while evaluating 'optionalString' at /home/jon/.cache/nixpkgs-review/pr-79630/nixpkgs/lib/strings.nix:180:5, called from /home/jon/.cache/nixpkgs-review/pr-79630/nixpkgs/pkgs/games/dwarf-fortress/game.nix:63:8:
while evaluating 'makeSearchPathOutput' at /home/jon/.cache/nixpkgs-review/pr-79630/nixpkgs/lib/strings.nix:142:5, called from /home/jon/.cache/nixpkgs-review/pr-79630/nixpkgs/pkgs/games/dwarf-fortress/game.nix:13:13:
while evaluating 'makeSearchPath' at /home/jon/.cache/nixpkgs-review/pr-79630/nixpkgs/lib/strings.nix:122:5, called from /home/jon/.cache/nixpkgs-review/pr-79630/nixpkgs/lib/strings.nix:142:11:
while evaluating anonymous function at /home/jon/.cache/nixpkgs-review/pr-79630/nixpkgs/lib/strings.nix:123:32, called from undefined position:
while evaluating the derivation attribute 'name' at /home/jon/.cache/nixpkgs-review/pr-79630/nixpkgs/pkgs/games/dwarf-fortress/unfuck.nix:51:3:
attribute 'unfuckRelease' missing, at /home/jon/.cache/nixpkgs-review/pr-79630/nixpkgs/pkgs/games/dwarf-fortress/unfuck.nix:51:35
```

**Comment by shazow** [comment]
Yea, as I mentioned too. Works fine for .02 but no longer for .01. Any ideas how to work around this?

File: pkgs/games/dwarf-fortress/unfuck.nix [hunk 2]
```diff
@@ -33,6 +33,10 @@ let
       sha256 = "1kszkb1d1vll8p04ja41nangsaxb5lv4p3xh2jhmsmipfixw7nvz";
     };
     "0.47.01" = {
+      unfuckrelease = "0.47.01";
```
Comment (jonringer) [inline pkgs/games/dwarf-fortress/unfuck.nix]:
```suggestion
      unfuckRelease = "0.47.01";
```

to fix evaluation

Comment (jonringer) [reply pkgs/games/dwarf-fortress/unfuck.nix]: r -> R

Comment (shazow) [reply pkgs/games/dwarf-fortress/unfuck.nix]: Aha!! Good catch, on my phone right now but I'll confirm tonight.

**Comment by shazow** [comment]
Pushed the update, looks better now.

**Review by jonringer — APPROVED** [review]
diff LGTM
commits LGTM

```
[25 built, 12 copied (392.9 MiB), 324.2 MiB DL]
https://github.com/NixOS/nixpkgs/pull/79630
15 package built:
dwarf-fortress-packages.cla-theme dwarf-fortress-packages.dwarf-fortress-full dwarf-fortress-packages.dwarf-fortress_0_47_02 dwarf-fortress-packages.phoebus-theme dwarf-fortress-packages.themes.afro-graphics dwarf-fortress-packages.themes.autoreiv dwarf-fortress-packages.themes.gemset dwarf-fortress-packages.themes.ironhand dwarf-fortress-packages.themes.jolly-bastion dwarf-fortress-packages.themes.obsidian dwarf-fortress-packages.themes.rally-ho dwarf-fortress-packages.themes.spacefox dwarf-fortress-packages.themes.taffer dwarf-fortress-packages.themes.tergel dwarf-fortress-packages.themes.wanderlust
```

**Comment by jonringer** [comment]
merging once ofborg is done with eval

**Pull request merged**